app/controllers/StorageExplorerController.scala

```diff
@@ -89,43 +95,62 @@ class StorageExplorerController @Inject() (
     future.map( s => Ok( Json.toJson( s ) ) )
 
   }
+  def getFiles( bucketid: Long, n: Option[Int], g: GraphTraversalSource ): GraphTraversal[Vertex, Vertex] = {
+    val nrfiles = n.getOrElse( 0 )
+    nrfiles match {
+      case 0 =>
+        logger.debug( "No file limit requested, returning all files in bucket " )
+        g.V( Long.box( bucketid ) ).in( "resource:stored_in" ).in( "resource:has_location" ).has( Constants.TypeKey, "resource:file" )
 
-  def fileList( id: Long ): Action[AnyContent] = ProfileFilterAction( jwtVerifier.get ).async { implicit request =>
-    logger.debug( "Request to retrieve all files in bucket with id " + id )
+      case x: Int =>
+        logger.debug( "File limit of " + x + " requested, returning " + x + " file(s) in bucket " )
+        g.V( Long.box( bucketid ) ).in( "resource:stored_in" ).in( "resource:has_location" ).has( Constants.TypeKey, "resource:file" ).tail( x )
```

Comment:
Limit, but no pagination?

Reply:
Pagination upon request...if it is needed I'll put it

Reply:
As a maintenance fix, I don't think we will need it.